authoringAssistant/src/main/java/explorableviz/transparenttext/Main.java

```diff
@@ -17,53 +17,74 @@ public static void main(String... args) {
-            final int queryLimit = numQueryToExecute.orElseGet(queryContexts::size);
-            final ArrayList<String> results = execute(learningQueryContext, arguments.get("agent"), queryLimit, queryContexts);
-            if (computeAccuracy(results, queryContexts, queryLimit, Integer.parseInt(arguments.get("threshold")))) {
+            Settings.init(arguments.get("settings"));
+            learningQuery = LearningQuery.importLearningCaseFromJSON(arguments.get("inContextLearningPath"), Integer.parseInt(arguments.get("numLearningCaseToGenerate")));
+            queries = TestQuery.loadCases(arguments.get("testPath"), Integer.parseInt(arguments.get("numTestToGenerate")));
+            final int queryLimit = numQueryToExecute.orElseGet(queries::size);
+            final ArrayList<QueryResult> results = execute(learningQuery, arguments.get("agent"), queryLimit, queries);
+            float accuracy = computeAccuracy(results, queries, queryLimit);
+            writeLog(results, arguments.get("agent"), learningQuery.size());
+            if (accuracy >= Float.parseFloat(arguments.get("threshold"))) {
+                System.out.println(STR."Accuracy OK =\{accuracy}");
                 System.exit(0);
             } else {
+                System.out.println(STR."Accuracy KO =\{accuracy}");
                 System.exit(1);
             }
         } catch (Exception e) {
             e.printStackTrace();
             System.exit(1);
         }
+
     }
 
-    private static boolean computeAccuracy(List<String> results, List<QueryContext> queryContexts, int queryLimit, float threshold) {
+    private static void writeLog(ArrayList<QueryResult> results, String agent, int learningContextSize) throws IOException {
+        Files.createDirectories(Paths.get(Settings.getLogFolder()));
+        PrintWriter out = new PrintWriter(new FileOutputStream(STR."\{Settings.getLogFolder()}/log_\{System.currentTimeMillis()}.csv"));
+        out.append("test-case;llm-agent;temperature;num-token;in-context-learning-size;attempts;result;generated-expression;expected;duration(ms)\n");
+
+        results.forEach(result -> {
+            out.append(STR."\{result.query().getTestCaseFileName()};");
+            out.append(STR."\{agent};");
+            out.append(STR."\{Settings.getTemperature()};");
+            out.append(STR."\{Settings.getNumContextToken()};");
+            out.append(STR."\{learningContextSize};");
+            out.append(STR."\{result.attempt()};");
+            out.append(STR."\{result.response() != null ? "OK" : "KO"};");
+            out.append(STR."\{result.response()};");
+            out.append(STR."\{result.query().getExpected()};");
+            out.append(STR."\{result.duration()};");
+            out.append("\n");
+        });
+
+        out.flush();
+        out.close();
+    }
+
+    private static float computeAccuracy(List<QueryResult> results, List<Query> queries, int queryLimit) {
         logger.info("Computing accuracy");
```

Comment:
Would it read better if `logger` was called `log`?

Reply:
I prefer logger, due the fact that is a kind of subject (agent) that do something (a log), and also because it's quite common in Java to call it logger.